barretenberg/cpp/src/barretenberg/vm/tests/helpers.test.cpp

```diff
@@ -39,8 +34,9 @@ void validate_trace_proof(std::vector<Row>&& trace)
  * @param trace Execution trace
  * @param selectRow Lambda serving to select the row in trace
  * @param newValue The value that will be written in intermediate register Ic at the selected row.
+ * @param alu A boolean telling whether we mutate the ic value in alu as well.
  */
-void mutate_ic_in_trace(std::vector<Row>& trace, std::function<bool(Row)>&& selectRow, FF const& newValue)
+void mutate_ic_in_trace(std::vector<Row>& trace, std::function<bool(Row)>&& selectRow, FF const& newValue, bool alu)
```

Comment:
is this used anywhere or is this for future failure case tests?

Reply:
This is already used for negative tests over FF. Now, I added other uses of it for the non'FF negative tests.